lib/chef/provisioning/aws_driver/driver.rb

```diff
@@ -53,6 +53,24 @@ class LoadBalancer
 end
 end
 
+require 'chef/provider/load_balancer'
+class Chef
+class Provider
+  class LoadBalancer
+    # We override this so we can specify a machine name as `i-123456`
+    # This is totally a hack until we move away from base resources
+    def get_machine_spec!(machine_name)
```

Comment:
In general, given the base resources and the poverty of `aws_instance`, this doesn't seem like a bad solution.

We could add a global flag for this, like `enable_instance_id_resolution` or something. That way users know they're adding magic.